test/e2e/test-e2e.js

```diff
@@ -0,0 +1,323 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+var assert = require('assert');
+var util = require('util');
+var _ = require('lodash'); // for _.find. Can't use ES6 yet.
+var cp = require('child_process');
+var semver = require('semver');
+var promisifyAll = require('@google-cloud/common').util.promisifyAll;
+var Debug = require('../..');
+var Debugger = require('../debugger.js');
+
+var CLUSTER_WORKERS = 3;
+
+var FILENAME = 'test/fixtures/fib.js';
+
+var delay = function(delayTimeMS) {
+  return new Promise(function(resolve, reject) {
+    setTimeout(resolve, delayTimeMS);
+  });
+};
+
+describe('@google-cloud/debug end-to-end behavior (allow 60s)', function () {
+  var api;
+
+  var debuggeeId;
+  var projectId;
+  var children = [];
+
+  before(function() {
+    promisifyAll(Debugger);
+    api = new Debugger(new Debug({}));
+  });
+
+  beforeEach(function() {
+    this.timeout(10 * 1000);
+    return new Promise(function(resolve, reject) {
+      var numChildrenReady = 0;
+      var handler = function(a) {
+        if (a[0].length > 0) {
+          reject(new Error('A child reported the following error: ' + a[0]));
+        }
+        if (!debuggeeId) {
+          // Cache the needed info from the first worker.
+          debuggeeId = a[1];
+          projectId = a[2];
+        } else {
+          // Make sure all other workers are consistent.
+          assert.equal(debuggeeId, a[1]);
+          assert.equal(projectId, a[2]);
+          if (debuggeeId !== a[1] || projectId !== a[2]) {
+            reject(new Error('Child debuggee ID and/or project ID' +
+                             'is not consistent with previous child'));
+          }
+        }
+        numChildrenReady++;
+        if (numChildrenReady === CLUSTER_WORKERS) {
+          resolve();
+        }
+      };
+
+      var stdoutHandler = function(index) {
+        return function(chunk) {
+          children[index].transcript += chunk;
+        };
+      };
+
+      for (var i = 0; i < CLUSTER_WORKERS; i++) {
+        // Fork child processes that communicate with this process with IPC.
+        var child = { transcript: '' };
+        child.process = cp.fork(FILENAME, {
+          execArgv: [],
+          env: {
+            GCLOUD_PROJECT: process.env.GCLOUD_PROJECT,
+            HOME: process.env.HOME
+          },
+          stdio: ['pipe', 'pipe', 'pipe', 'ipc']
+        });
+        child.process.on('message', handler);
+
+        children.push(child);
+
+        // Each child has its own transcript.
+        child.process.stdout.on('data', stdoutHandler(i));
+        child.process.stderr.on('data', stdoutHandler(i));
+      }
+    });
+  });
+
+  afterEach(function() {
+    children.forEach(function (child) {
+      child.process.kill();
```

Comment:
I think it's technically not possible for children to print error messages after the beforeEach hook is resolved, but this will reduce the possibility of tests interfering with one another.